### PR TITLE
Added sorting implementations.

### PR DESCRIPTION
**Pull request by JoseLlorensRipolles**
Added Insertion, Selection and Quicksort implementations. Renamed Mergesort  to MergeSort and changed nmethod name and declaration to fit the new sorting interface. Added a new sorting interface for all sortring implementations. Changed to non-static methods.

### CODE REVIEW

File: java/src/eda/sorting/InsertionSort.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        for(int i = 1;i<arr.length;i++){
+            aux=arr[i];
+            int j;
+            for(j = i-1;j>=0 && arr[j].compareTo(aux)>0;j--){
+                arr[j+1]=arr[j];
+            }
+            arr[j+1]=aux;
+        }
+    }
+    public static void main(String[]args){
```
Comment (tonilopezmr) [inline java/src/eda/sorting/InsertionSort.java]: Remove main method and create a test like ExamMarks.

File: java/src/eda/sorting/InsertionSort.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package eda.sorting;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+/**
+ * Created by JoseManuel on 19/04/2016.
+ */
+public class InsertionSort<T extends Comparable<T>> implements Sorting<T> {
```
Comment (tonilopezmr) [inline java/src/eda/sorting/InsertionSort.java]: If InsertionSort methods are statics, You don't need create generic class, only in the methods declaration.

File: java/src/eda/sorting/Sorting.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package eda.sorting;
+
+/**
+ * Created by JoseManuel on 19/04/2016.
+ */
+public interface Sorting<T extends Comparable<T>> {
+    public static<T extends Comparable<T>> void sort(T[] arr){
```
Comment (tonilopezmr) [inline java/src/eda/sorting/Sorting.java]: When you create a interface, the methods can't be statics. The interface methods can't be implemented.

**Comment by JoseLlorensRipolles** [comment]
Hey! I've already implemented the sorting tests! Take a look and give me feedback!

File: java/test/eda/sorting/SortingTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+public class SortingTest {
+
+    private Sorting sortingTest;
+
+    public SortingTest(Sorting sortingTest) {
+        this.sortingTest = sortingTest;
+    }
+
+    @Test
+    public void test_random(){
```
Comment (tonilopezmr) [inline java/test/eda/sorting/SortingTest.java]: All method names must be start with *should_*  for example: should_do_something_when_other_thing....

File: java/test/eda/sorting/SortingTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        Integer[] h = new Integer[50000];
+        for(int i = 0; i<h.length;i++){
```
Comment (tonilopezmr) [inline java/test/eda/sorting/SortingTest.java]: You always create a random array and we can't see what is the expected array, aren't you? 
I think that if you only create a one method with random array with hight length is enough. 
Each other method should creates a -manual- array and We can see that the array {1,4,3} finally is sorted like this {1,3,4}.

But well played!!

**Comment by JoseLlorensRipolles** [comment]
Done! :+1:

**Comment by JoseLlorensRipolles** [comment]
Correted the things you've said! Check it now please!

**Comment by tonilopezmr** [comment]
Thanks to contribute to #3 !!!

**Pull request merged**